Sources/SkipUI/SkipUI/Text/Label.swift

```diff
@@ -46,35 +46,41 @@ public struct Label : View {
-        if placement.contains(ViewPlacement.toolbar) {
-            ComposeImage(context: context)
-        } else if placement.contains(ViewPlacement.systemTextColor) && !EnvironmentValues.shared.isEnabled && EnvironmentValues.shared._foregroundStyle == nil {
-            ComposeLabel(context: context, imageColor: Color.primary.opacity(Double(ContentAlpha.disabled)))
+        if style == .automatic {
+            if placement.contains(ViewPlacement.toolbar) {
+                style = .iconOnly
+            } else {
+                style = .titleAndIcon
+            }
+        }
+        var imageColor: Color?
+        var titlePadding = 0.0
+        if placement.contains(ViewPlacement.systemTextColor) && !EnvironmentValues.shared.isEnabled && EnvironmentValues.shared._foregroundStyle == nil {
+            imageColor = Color.primary.opacity(Double(ContentAlpha.disabled))
         } else if placement.contains(ViewPlacement.onPrimaryColor) && EnvironmentValues.shared._foregroundStyle == nil {
             var imageColor = Color(colorImpl: { MaterialTheme.colorScheme.onPrimary })
             if !EnvironmentValues.shared.isEnabled {
                 imageColor = imageColor.opacity(Double(ContentAlpha.disabled))
             }
-            ComposeLabel(context: context, imageColor: imageColor)
         } else if placement.contains(ViewPlacement.listItem) {
-            ComposeLabel(context: context, imageColor: EnvironmentValues.shared._foregroundStyle as? Color ?? EnvironmentValues.shared._listItemTint ?? Color.accentColor, titlePadding: 6.0)
-        } else {
-            ComposeLabel(context: context)
+            imageColor = EnvironmentValues.shared._foregroundStyle as? Color ?? EnvironmentValues.shared._listItemTint ?? Color.accentColor
+            titlePadding = 6.0
+        }
+        switch style {
+        case .titleOnly:
+            ComposeTitle(context: context)
+        case .iconOnly:
+            ComposeImage(context: context, imageColor: imageColor)
+        default:
+            ComposeLabel(context: context, imageColor: imageColor, titlePadding: titlePadding)
         }
     }
 
-    @Composable private func ComposeLabel(context: ComposeContext, imageColor: Color? = nil, titlePadding: Double = 0.0) {
+    @Composable private func ComposeLabel(context: ComposeContext, imageColor: Color?, titlePadding: Double) {
         Row(modifier: context.modifier, horizontalArrangement: Arrangement.spacedBy(8.dp), verticalAlignment: androidx.compose.ui.Alignment.CenterVertically) {
-            if let imageColor {
-                EnvironmentValues.shared.setValues {
-                    $0.set_foregroundStyle(imageColor)
-                } in: {
-                    image.Compose(context: context.content())
-                }
-            } else {
-                image.Compose(context: context.content())
-            }
+            ComposeImage(context: context, imageColor: imageColor)
```

Comment:
I'd like to do context.content() here and use the given Context as-is in ComposeImage(), because ComposeImage() *should* apply the full context if called directly from the style switch or is called by some other component (it's not private)

Reply:
Oh, one other thing: it'd be great if you added .labelStyle support to the supported API table in the README